### PR TITLE
Util: a optimization of mysql.Bit in Datum

### PR DESCRIPTION
**Pull request by zxylvlp**
We can use the space of Datum to save mysql.Bit to save the cost of interface.

### CODE REVIEW

**Comment by zxylvlp** [comment]
@coocood @shenli

File: util/types/datum.go [hunk 1]
```diff
@@ -50,14 +50,17 @@ const (
 // Datum is a data box holds different kind of data.
 // It has better performance and is easier to use than `interface{}`.
 type Datum struct {
-	k int         // datum kind.
-	i int64       // i can hold int64 uint64 float64 values.
-	b []byte      // b can hold string or []byte values.
-	x interface{} // f hold all other types.
+	k   byte        // datum kind.
+	u8  uint8       // u8 can hold values in byte.
```
Comment (coocood) [inline util/types/datum.go]: rename the new fields to `collation`, `decimal` and `length`.

**Comment by shenli** [comment]
LGTM
But we should remember that there may be a potential performance issue.

**Comment by coocood** [comment]
LGTM

**Pull request merged**